src/lib.rs

```diff
@@ -119,7 +119,8 @@ impl GitBom {
     }
 
     pub fn add(&mut self, gitoid: String) {
-      self.gitOids.push(gitoid) 
+      self.gitOids.push(gitoid);
+      self.gitOids.sort();
```

Comment:
Why sort on every `push`?

Why do the oids need to be sorted?

Also, the `Vec<String>` should not be public... it should only be visible to the `struct`. With such a mechanism, if we need to sort the `git_oids`, we can do it when they are returned from an accessor function.

Reply:
There are two parts to this answer:

Spec wise:
1. GitBom spec itself calls for gitoids to be sorted in the produced document.

Given this, implementation wise:

1. sorting in add keeps the internal state consistent with expectations.
2. The above PR is a naive approach, happy to take or consider other paths.
3.  Short of using a hash table, makes it easier to find duplicates gitoids blob entries at add time. PR on this to come later.

Reply:
I've created this code to demonstrate your dual goals of de-duping oids and a facility for getting sorted oids: https://github.com/dpp/gitbom-rs/commit/aa615afc7c0922d8fb1dd642ac08b19f47da5a8d

There are two underlying concerns I have with the approach you've takes:

1. It makes the `add` function O(n^2 log n)... if one adds n items, each add will require an O(n log n) to sort. Using sets means the cost of an `add` is O(log n) so all the adds become O(n log n)
2. I'm strongly biasing to using immutable/persistent data structures. I'll start a thread on Slack for the reasoning

It may also be possible to use a sorted set, but I could not find any sorted set libraries that also have O(1) `clone()` operations where the `im` library (which is based on [HAMTs](https://en.wikipedia.org/wiki/Hash_array_mapped_trie))
has a lot of real-world benefits as demonstrated in the Scala and Clojure libraries.

Reply:
definitely agree, the initial goal for me is to get something working then optimize the underlying data structures. happy to see these exist in rust out of the box. closing this.